TrackingBackend/app/utils/misc_utils.py

```diff
@@ -0,0 +1,29 @@
+from app.types import EyeData, TRACKING_FAILED
+from queue import Queue, Empty
+from cv2.typing import MatLike
+
+
+def clamp(x, low, high):
+    return max(low, min(x, high))
+
+
+def clear_queue(queue: Queue) -> None:
+    while True:
+        try:
+            queue.get(block=True, timeout=0.1)
+        except Empty:
+            break
+
+
+# Base class for all algorithms
+class BaseAlgorithm:
+    # all algorithms must implement this method
+    def run(self, frame: MatLike) -> EyeData:
+        return TRACKING_FAILED
+
+    def normalize(self, x: float, y: float, width: int, height: int) -> tuple[float, float]:
+        """takes a point and normalizes it to a range of 0 to 1"""
+        tx: float = x / width
+        ty: float = y / height
+
+        return tx, ty
```

Comment:
I feel like this doesn't really belong in utils, we might want to move this out eventually

With the config rewrite I just made a `base_module` file, in config package, that only described the interface and every module implementing it was placed in a separate file next to it as well 

What do you think? We're gonna have a couple of algos so grouping them might be worth it

Reply:
maybe?
i dont fully understand what you mean by grouping them since they are already all in one folder, where each file is its own algorithm, i do think we should move this base class somewhere else

Reply:
Ah, sorry, I was mostly going off of github thingy so I didn't notice, but you're right xD